wordpress-mobile/gutenberg-mobile
# Flatten embedded frameworks when generating XCFramework

The goal of this PR is to flatten frameworks nested under the `Frameworks` folder of other frameworks. Especially, the Hermes framework which is nested under the Gutenberg framework.

This has been achieved by simply moving the embedded frameworks to the root level (https://github.com/wordpress-mobile/gutenberg-mobile/pull/5968/commits/e03356d4bb3cba27c5e5b7448caa814a11d8bd47), plus including the Hermes framework in the vendor framework list of the Gutenberg `podspec` (https://github.com/wordpress-mobile/gutenberg-mobile/pull/5968/commits/355f4319c3be6e31fc509d42bb41698078385d62).

## To test

**Preparation:**
* Check out the branch `gutenberg/tweak-pod-deps-by-rn-version` from the WP-iOS repository ([PR reference](https://github.com/wordpress-mobile/WordPress-iOS/pull/21021)).

### Test via CI
* Wait until `Build iOS RN XCFramework & Publish to S3` CI job finishes.
* Update the Gutenberg Mobile reference to the commit `64a5fed3fd28e7fa4aa25edc78f39b76617e5f71`.
* Run the command `bundle install && bundle exec pod install`.
  **NOTE:** If any error is encountered, try to run the command `rm -rf build && rm -rf Pods` before installing Pods.
* Build and run the WP-iOS app using a physical device.
* Observe that the app doesn't crash and that the editor works as expected.

### Test via local installation

⚠️ Note that this testing approach is mainly to test XCFramework integration without having to publish the framework.

* Apply the following patch:
```
diff --git forkSrcPrefix/ios-xcframework/Gutenberg.podspec forkDstPrefix/ios-xcframework/Gutenberg.podspec
new file mode 100644
index 0000000000000000000000000000000000000000..ccba32ed4e4053b51f638d3238e4213324f34ca9
--- /dev/null
+++ forkDstPrefix/ios-xcframework/Gutenberg.podspec
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+Pod::Spec.new do |s|
+  s.name = 'Gutenberg'
+  s.summary = 'Printing since 1440'
+  s.homepage = 'https://github.com/wordpress-mobile/gutenberg-mobile'
+  s.authors = 'Automattic'
+
+  s.version = '1.98.1'
+
+  s.license = { type: 'GPL-2' }
+
+  s.ios.deployment_target = '15.0'
+  s.swift_version = '5.0'
+
+  s.source = {
+    http: 'file://' + __dir__ + '/build/xcframeworks/Gutenberg.tar.gz'
+  }
+
+  s.vendored_frameworks = [
+    'Frameworks/Aztec.xcframework',
+    'Frameworks/Gutenberg.xcframework',
+    'Frameworks/React.xcframework',
+    'Frameworks/RNTAztecView.xcframework',
+    'Frameworks/yoga.xcframework',
+    'Frameworks/hermes.xcframework'
+  ]
+end
```
* In the WP-iOS project, change [the Gutenberg Mobile `podspec` reference](https://github.com/wordpress-mobile/WordPress-iOS/blob/636d972c239d15818166fd06b7ee417b55d99fe2/Gutenberg/cocoapods_helpers.rb#L45) to: `../gutenberg-mobile/ios-xcframework/Gutenberg.podspec`
* Run the command `bundle install && bundle exec pod install`.
  **NOTE:** If any error is encountered, try to run the command `rm -rf build && rm -rf Pods` before installing Pods.
* Build and run the WP-iOS app using a physical device.
* Observe that the app doesn't crash and that the editor works as expected.

PR submission checklist:
